### PR TITLE
Fixed: Skip github build status notifications

### PR DESCRIPTION
**Pull request by gaffeyQiu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it:**
Create github pipeline use Skip Notifications Trait plugin forbid github plugin send status notification (also include gitlab, bitbucket)

**Which issue(s) this PR fixes:**
#804

### Does this PR introduce a user-facing change??

```release-note
None
```

### CODE REVIEW

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
How about other git providers (gitlab, bitbucket)? And we need to consider both the reading and writing process.

By the way, the PR template is meaningful. It's necessary to follow the template.

**Comment by gaffeyQiu** [comment]
skip-notifications-trait plugin support github, bitbucket, gitlab support is not written into the plugin documentation, But I just tested is effective，Can I continue to use this plugin support gitlab， bitbucket?
---
MR1 is not use skip-notifications-trait plugin

**Comment by LinuxSuRen** [comment]
Why closed?

**Comment by gaffeyQiu** [comment]
Did I make a mistake？ I was just modifying the ks-jenkins repo pull request

**Comment by gaffeyQiu** [comment]
> Why closed?

Sorry, Please reopen it

**Comment by gaffeyQiu** [comment]
Reopen it

File: pkg/client/devops/jenkins/internal/gitlab.go [hunk 1]
```diff
@@ -86,6 +86,7 @@ func AppendGitlabSourceToEtree(source *etree.Element, gitSource *devopsv1alpha3.
 		regexTraits.CreateAttr("plugin", "scm-api")
 		regexTraits.CreateElement("regex").SetText(gitSource.RegexFilter)
 	}
+	traits.CreateElement("io.jenkins.plugins.gitlabbranchsource.GitLabSkipNotificationsTrait")
```
Comment (LinuxSuRen) [inline pkg/client/devops/jenkins/internal/gitlab.go:89]: I prefer to provide an option for this. So, users are able to choose if skip the notification from Jenkins by themself. We could make this to be the default behavior.

For example, `gitSource.SkipJenkinsNotification`.

Comment (gaffeyQiu) [reply pkg/client/devops/jenkins/internal/gitlab.go:89]: Yes, good advice.

Comment (gaffeyQiu) [reply pkg/client/devops/jenkins/internal/gitlab.go:89]: @LinuxSuRen  I want define `SkipJenkinsNotification` field is `bool`, but `bool` default is `false`.  
Change field name `AcceptJenkinsNotification` ?

Comment (LinuxSuRen) [reply pkg/client/devops/jenkins/internal/gitlab.go:89]: Fine for me.

Comment (LinuxSuRen) [reply pkg/client/devops/jenkins/internal/gitlab.go:89]: By the way, please don't forget to add the unit tests.

**Comment by gaffeyQiu** [comment]
PR1 is not use skip-notifications-trait plugin
PR2 is use skip-notifications-trait plugin
image
image
So, I need add this line to config.xml
image

**Comment by gaffeyQiu** [comment]
Why the codecov is 0%? 
`Test_MultiBranchPipelineSkipNotification()` It should be covered `if !gitSource.AcceptJenkinsNotification { xxx }`  case
image

**Review by LinuxSuRen — COMMENTED** [review]
One more thing that you need to do:

```shell
make manifests generate
```

This command will generate the CRD YAML files.

**Comment by gaffeyQiu** [comment]
I run `make manifests generate` generat an old version CRD, because my controller-gen is the old version, advice add validation in makefile
image

**Comment by LinuxSuRen** [comment]
> Why the codecov is 0%?

I guess there are different coverage methods.

image

**Comment by LinuxSuRen** [comment]
> And if you want to indicate the specified version of controller-gen you can refer to the practice [here](https://github.com/fluxcd/helm-controller/blob/12dc7d9dcab039bfbf8a555c825a489f3863848e/Makefile#L116).

Please feel free to help us improve it if you're interested in it.

**Review by LinuxSuRen — APPROVED** [review]
I have tested it with the following image manually. It works well. Thanks for your effort @gaffeyQiu , good job!

```
kubespheredev/devops-controller:dev-v3.3.1-rc.1-2629488
```

image

image

**Comment by LinuxSuRen** [comment]
The related PR is https://github.com/kubesphere/ks-jenkins/pull/91

**Comment by gaffeyQiu** [comment]
> I have tested it with the following image manually. It works well. Thanks for your effort @gaffeyQiu , good job!
> 
> ```
> kubespheredev/devops-controller:dev-v3.3.1-rc.1-2629488
> ```
> 
> image
> 
> image

It was a pleasure to contribute

**Pull request merged**